dpnp/dpnp_iface_linearalgebra.py

```diff
@@ -183,36 +230,323 @@ def einsum(*args, **kwargs):
+    >>> np.einsum("k...,jk", a, b)
+    array([[10, 28, 46, 64],
+           [13, 40, 67, 94]])
+
+    Chained array operations. For more complicated contractions, speed ups
+    might be achieved by repeatedly computing a "greedy" path or pre-computing
+    the "optimal" path and repeatedly applying it, using an `einsum_path`
+    insertion. Performance improvements can be particularly significant with
+    larger arrays:
+
+    >>> a = np.ones(64000).reshape(20, 40, 80)
+
+    Basic `einsum`: 119 ms ± 26 ms per loop (benchmarked on 12th
+    Gen Intel\u00AE Core\u2122 i7 processor)
+
+    >>> %timeit np.einsum("ijk,ilm,njm,nlk,abc->",a,a,a,a,a)
+
+    Sub-optimal `einsum`: 32.9 ms ± 5.1 ms per loop
+
+    >>> %timeit np.einsum("ijk,ilm,njm,nlk,abc->",a,a,a,a,a, optimize="optimal")
+
+    Greedy `einsum`: 28.6 ms ± 4.8 ms per loop
+
+    >>> %timeit np.einsum("ijk,ilm,njm,nlk,abc->",a,a,a,a,a, optimize="greedy")
+
+    Optimal `einsum`: 26.9 ms ± 6.3 ms per loop
+
+    >>> path = np.einsum_path(
+        "ijk,ilm,njm,nlk,abc->",a,a,a,a,a, optimize="optimal"
+    )[0]
+    >>> %timeit np.einsum("ijk,ilm,njm,nlk,abc->",a,a,a,a,a, optimize=path)
 
     """
 
-    return call_origin(numpy.einsum, *args, **kwargs)
+    if optimize is True:
+        optimize = "greedy"
+
+    return dpnp_einsum(
+        *operands,
+        out=out,
+        dtype=dtype,
+        order=order,
+        casting=casting,
+        optimize=optimize,
+    )
 
 
-def einsum_path(*args, **kwargs):
+def einsum_path(*operands, optimize="greedy", einsum_call=False):
     """
-    einsum_path(subscripts, *operands, optimize='greedy')
+    einsum_path(subscripts, *operands, optimize="greedy")
 
     Evaluates the lowest cost contraction order for an einsum expression
     by considering the creation of intermediate arrays.
 
     For full documentation refer to :obj:`numpy.einsum_path`.
 
-    Limitations
-    -----------
-    Function is executed sequentially on CPU.
+    Parameters
+    ----------
+    subscripts : str
+        Specifies the subscripts for summation.
+    *operands : sequence of arrays
+        These are the arrays for the operation in any form that can be
+        converted to an array. This includes scalars, lists, lists of
+        tuples, tuples, tuples of tuples, tuples of lists, and ndarrays.
+    optimize : {bool, list, tuple, None, "greedy", "optimal"}
+        Choose the type of path. If a tuple is provided, the second argument is
+        assumed to be the maximum intermediate size created. If only a single
+        argument is provided the largest input or output array size is used
+        as a maximum intermediate size.
+
+        * if a list is given that starts with ``einsum_path``, uses this as the
+          contraction path
+        * if ``False`` or ``None`` no optimization is taken
+        * if ``True`` defaults to the "greedy" algorithm
+        * ``"optimal"`` is an algorithm that combinatorially explores all
```

Comment:
It's required to extend `doc/known_words.txt` with `combinatorially` word or rephrase.